org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverter_23Test.java

```diff
@@ -133,4 +127,404 @@ void main() {
+	    TypeDeclaration a = (TypeDeclaration) compilationUnit.types().get(0);
+
+	    assertTrue(Modifier.isProtected(a.getModifiers()));
+	    assertTrue(Modifier.isAbstract(a.getModifiers()));
+
+	    ITypeBinding aBinding = a.resolveBinding();
+
+	    assertTrue(Modifier.isProtected(aBinding.getModifiers()));
+	    assertTrue(Modifier.isAbstract(aBinding.getModifiers()));
+	}
+
+	//public sealed interface
+	public void test003_k() throws CoreException {
+		ASTParser astParser = ASTParser.newParser(getAST23());
+	    Map<String, String> options = new HashMap<>();
+	    options.put(JavaCore.COMPILER_COMPLIANCE, "23");
+	    options.put(JavaCore.COMPILER_SOURCE, "23");
+
+	    astParser.setCompilerOptions(options);
+	    astParser.setEnvironment(new String[] {}, new String[] {}, new String[] {}, true);
+	    astParser.setUnitName("Example.java");
+	    astParser.setResolveBindings(true);
+	    astParser.setBindingsRecovery(true);
+
+	    String source ="""
+		    		public sealed interface A permits X {}
+	    			public final  class X implements A {}
+	    		""";
+
+	    astParser.setSource(source.toCharArray());
+
+	    CompilationUnit compilationUnit = (CompilationUnit) astParser.createAST(null);
+	    TypeDeclaration a = (TypeDeclaration) compilationUnit.types().get(0);
+
+	    assertTrue(Modifier.isSealed(a.getModifiers()));
+	    assertTrue(Modifier.isPublic(a.getModifiers()));
+	    assertEquals("permitted types are not present in AST", a.permittedTypes().size(), 1);
+
+	    ITypeBinding aBinding = a.resolveBinding();
+	    assertTrue(Modifier.isSealed(aBinding.getModifiers()));
+	    assertTrue(Modifier.isPublic(aBinding.getModifiers()));
+	}
+
+	//public non-sealed interface
+	public void test003_l() throws CoreException {
+		ASTParser astParser = ASTParser.newParser(getAST23());
+	    Map<String, String> options = new HashMap<>();
+	    options.put(JavaCore.COMPILER_COMPLIANCE, "23");
+	    options.put(JavaCore.COMPILER_SOURCE, "23");
+
+	    astParser.setCompilerOptions(options);
+	    astParser.setEnvironment(new String[] {}, new String[] {}, new String[] {}, true);
+	    astParser.setUnitName("Example.java");
+	    astParser.setResolveBindings(true);
+	    astParser.setBindingsRecovery(true);
+
+	    String source ="""
+		    		public non-sealed interface A permits X {}
+	    			public final  class X implements A {}
+	    		""";
+
+	    astParser.setSource(source.toCharArray());
+
+	    CompilationUnit compilationUnit = (CompilationUnit) astParser.createAST(null);
+	    TypeDeclaration a = (TypeDeclaration) compilationUnit.types().get(0);
+
+	    assertTrue(Modifier.isNonSealed(a.getModifiers()));
+	    assertTrue(Modifier.isPublic(a.getModifiers()));
+	    assertEquals("permitted types are not present in AST", a.permittedTypes().size(), 1);
+
+	    ITypeBinding aBinding = a.resolveBinding();
+	    assertTrue(Modifier.isNonSealed(aBinding.getModifiers()));
```

Comment:
Try adding a test here : `assertFalse(Modifier.isSealed(aBinding.getModifiers()));`
This should not fail, but it does with the patch.